airbyte_cdk/sources/declarative/decoders/composite_raw_decoder.py

```diff
@@ -35,15 +35,36 @@ def parse(
 class GzipParser(Parser):
     inner_parser: Parser
 
+    def _reset_reader_pointer(self, data: BufferedIOBase) -> None:
+        """
+        Reset the reader pointer to the beginning of the data.
+
+        Note:
+            - This is necessary because the gzip decompression will consume the data stream.
+        """
+        data.seek(0)
+
     def parse(
         self,
         data: BufferedIOBase,
     ) -> Generator[MutableMapping[str, Any], None, None]:
         """
         Decompress gzipped bytes and pass decompressed data to the inner parser.
+
+        IMPORTANT:
+            - If the data is not gzipped, reset the pointer and pass the data to the inner parser as is.
+
+        Note:
+            - The data is not decoded by default.
         """
-        with gzip.GzipFile(fileobj=data, mode="rb") as gzipobj:
-            yield from self.inner_parser.parse(gzipobj)
+
+        try:
```

Comment:
Should we add a test for this new logic?

Reply:
This logic is no longer present, the rest is covered with either existing tests or newly added one in:
- `unit_tests/sources/declarative/decoders/test_composite_decoder.py`
   - `test_composite_raw_decoder_gzip_jsonline_parser_decodes_non_gzipped_raw_response`